### ISSUE TITLE
Node count issue

### ISSUE THREAD

**Issue by sarath-s-kumar** [terraform-google-modules/terraform-google-kubernetes-engine#1580]
### TL;DR

If i enabled autoscaler for a node and using total limits per zone options then we configured the node count is total min count is 3 and total max count is 80. In the same node config we configured the initial node count as 10 then when we trigger the cluster will take only the total min node count in the number of nodes. Then what's the use for initial_node_count or do we need to configure this initial node count variable in the cluster level.

### Expected behavior

If i use the initial node count as 10 then it will create 10 nodes per zone as number of nodes

### Observed behavior

It will take the total min node count as number of nodes per zone

### Terraform Configuration

```hcl
# google_client_config and kubernetes provider must be explicitly specified like the following.

module "gke_controlplane_labels" {
    source = "git::https://github.com/Org-name/terraform-google-tagging.git//modules/labels_global?ref=v1.0.0"
    tier = "t1"
    organisation = var.organisation
    environment = "prod"
    region = var.region
    datacenter = var.datacenter

    extra_labels = {
      pod = "order"
      service_name = "gke-cluster"
      function = "gke"
      role = "controlplane"
      purpose = "order"
    }
}

# Seperating labels as change in nodepool labels force replaces it's creation.
module "gke_nodepool_labels" {
    source = "git::https://github.com/org-name/terraform-google-tagging.git//modules/labels_global?ref=v1.0.0"
    tier = "t1"
    organisation = var.organisation
    environment = "prod"
    region = var.region
    datacenter = var.datacenter

    extra_labels = {
      pod = "order"
      sub-pod = "order"
      org = "order"
      team = "order"
      service_name = "gke-nodepool"
      function = "gke"
      role = "nodepool"
      purpose = "order"
    }
}




module "gke" {
  source                     = "git::https://github.com/org-name/terraform-google-kubernetes-engine.git//modules/beta-private-cluster?ref=v1.24.9-gke"
  project_id                 = var.service_project_id
  name                       = var.gke_cluster_name
  region                     = var.region
  zones                      = ["asia-south1-a", "asia-south1-b", "asia-south1-c"]
  network                    = var.network_name
  network_project_id         = var.host_project_id
  subnetwork                 = var.subnetwork_name
  ip_range_pods              = var.subnet_ip_range_pods
  ip_range_services          = var.subnet_ip_range_services
  http_load_balancing        = true
  network_policy_provider    = "PROVIDER_UNSPECIFIED"
  datapath_provider = "ADVANCED_DATAPATH"
  enable_network_egress_export = true
  enable_cost_allocation = true
  enable_resource_consumption_export = true
  enable_shielded_nodes      = true
  grant_registry_access = false
  horizontal_pod_autoscaling = true
  filestore_csi_driver       = false
  enable_private_endpoint    = true
  enable_private_nodes       = true
  master_ipv4_cidr_block     = var.controlplane_cidr_block
  master_authorized_networks = var.controlplane_authorized_networks
  remove_default_node_pool   = true
  master_global_access_enabled = true
  release_channel       = "UNSPECIFIED"
  kubernetes_version = "1.24.9-gke.1500"
  resource_usage_export_dataset_id = "billing_production"
  
  // The desired Pub/Sub topic to which notifications will be sent by GKE. Format is projects/{project}/topics/{topic}.
  notification_config_topic = var.notification_topic_id

  cluster_autoscaling = {
    enabled             = false
    max_cpu_cores       = 0
    min_cpu_cores       = 0
    max_memory_gb       = 0
    min_memory_gb       = 0
    gpu_resources       = []
    autoscaling_profile = "OPTIMIZE_UTILIZATION"
  }

  maintenance_start_time = "2022-04-01T01:00:00Z"
  maintenance_end_time = "2022-04-02T01:00:00Z"
  maintenance_recurrence = "FREQ=WEEKLY;BYDAY=FR"
  cluster_resource_labels = module.gke_controlplane_labels.labels
  node_pools = [
    {
      name                      = "order-node-pool-001"
      machine_type              = "n2-custom-32-40960"
      node_locations            = "asia-south1-a,asia-south1-b,asia-south1-c"
      total_min_count           = 3
      total_max_count           = 80
      local_ssd_count           = 0
      spot                      = false
      local_ssd_ephemeral_count = 0
      disk_size_gb              = 100
      disk_type                 = "pd-standard"
      image_type                = "COS_CONTAINERD"
      auto_repair               = true
      auto_upgrade              = true
      service_account           = var.gke_nodepool_sa
      preemptible               = false
      initial_node_count        = 10
      max_pods_per_node         = 16
      enable_secure_boot = true
    },
  ]
  
  
  node_pools_oauth_scopes = {
    all = [
      "https://www.googleapis.com/auth/logging.write", # For logging
      "https://www.googleapis.com/auth/monitoring", # For monitoring
      "https://www.googleapis.com/auth/cloud-platform" # For image-pullsecrets
    ]
    
    order-node-pool-001 = []
   
  }

  node_pools_labels = {
    all = module.gke_nodepool_labels.labels
    
    order-node-pool-001 = {
      order-node-pool-001 = true
      pool-type = "order-node-pool-001"
    }
    
  }

  node_pools_metadata = {
    all = module.gke_nodepool_labels.labels

    order-node-pool-001 = {
      node-pool-metadata-custom-value = "order-node-pool-001"
    }
    
  }

  node_pools_taints = {
    all = []

  }

  node_pools_tags = {
    all = []

    order-node-pool-001 = [
      "order-node-pool-001",
    ] 
  }
}
```

### Terraform Version

```sh
Terraform v1.3.7
```
